rabix-engine/src/main/java/org/rabix/engine/jdbi/impl/JDBIIntermediaryFilesRepository.java

```diff
@@ -0,0 +1,77 @@
+package org.rabix.engine.jdbi.impl;
+
+import java.lang.annotation.Annotation;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
+import org.rabix.engine.jdbi.impl.JDBIIntermediaryFilesRepository.IntermediaryFileEntityMapper;
+import org.rabix.engine.repository.IntermediaryFilesRepository;
+import org.skife.jdbi.v2.SQLStatement;
+import org.skife.jdbi.v2.StatementContext;
+import org.skife.jdbi.v2.sqlobject.Bind;
+import org.skife.jdbi.v2.sqlobject.Binder;
+import org.skife.jdbi.v2.sqlobject.BinderFactory;
+import org.skife.jdbi.v2.sqlobject.BindingAnnotation;
+import org.skife.jdbi.v2.sqlobject.SqlQuery;
+import org.skife.jdbi.v2.sqlobject.SqlUpdate;
+import org.skife.jdbi.v2.sqlobject.customizers.RegisterMapper;
+import org.skife.jdbi.v2.tweak.ResultSetMapper;
+import org.skife.jdbi.v2.unstable.BindIn;
+
+
+@RegisterMapper(IntermediaryFileEntityMapper.class)
+public interface JDBIIntermediaryFilesRepository extends IntermediaryFilesRepository {
+  
+  @SqlUpdate("insert into intermediary_files (root_id,filename,count) values (:root_id,:filename,:count)")
+  void insert(@Bind("root_id") UUID root_id, @Bind("filename") String filename, @Bind("count") Integer count);
+  
+  @SqlUpdate("update intermediary_files set count=:count where root_id=:root_id and filename=:filename")
+  void update(@Bind("root_id") UUID root_id, @Bind("filename") String filename, @Bind("count") Integer count);
+  
+  @SqlUpdate("delete from intermediary_files where root_id=:root_id and filename=:filename")
+  void delete(@Bind("root_id") UUID root_id, @Bind("filename") String filename);
+  
+  @SqlUpdate("delete from intermediary_files where root_id=:root_id")
+  void delete(@Bind("root_id") UUID root_id);
+  
+  @Override
+  @SqlUpdate("delete from intermediary_files where root_id in (<ids>)")
```

Comment:
In order to run IN clauses, anotate interface with the **@UseStringTemplate3StatementLocator**